### ISSUE TITLE
False positives in upcoming ﻿Performance/StringReplacement cop

### ISSUE THREAD

**Issue by DMA57361** [bbatsov/rubocop#2061]
I've just been having a play with the master branch (to see how the todo generation now builds an exclude list - looks like that feature is going to be very handy!) and noticed a problem with the new Performance/StringReplacement cop.

Simple example code:

    puts 'Hello, World'.gsub(/lo/, 'p!')

With just this cop and auto-correct enabled Rubocop says:

    example.rb:1:21: C: [Corrected] Performance/StringReplacement: Use tr instead of gsub.
    puts 'Hello, World'.gsub('lo', 'p!')

And the corrected version is

    puts 'Hello, World'.tr('lo', 'p!')

However, output for these is very different:

    2.2.0 :001 > puts 'Hello, World'.gsub(/lo/, 'p!')
    Help!, World
     => nil
    2.2.0 :002 > puts 'Hello, World'.tr('lo', 'p!')
    Hepp!, W!rpd
     => nil
    2.2.0 :003 >

`gsub` replaces each `lo` with `p!`.
`tr` _independently_ replaces each `l` with `p` and each `o` with `!`.
I suspect what you're trying to do here is only safe when the regex used by gsub is a single character.

**Comment by bbatsov** [comment]
Definitely a bug. //cc @rrosenblum

**Comment by rrosenblum** [comment]
Good catch @DMA57361. I am glad this was found before a new tag was created.

**Comment by Rockyspade** [comment]
No false pretenses

**Comment by chamnap** [comment]
I still faced this issue too with the latest version. Is it fixed yet?

**Comment by pocke** [comment]
@chamnap This offense is correct.

```ruby
p 'foo/bar/baz'.tr('/', '_')      # => "foo_bar_baz"
p 'foo/bar/baz'.gsub(%r{\/}, '_') # => "foo_bar_baz"
```

`String#tr` does not receive a regular expression, however, if the regular expression's length is one, we can replace the regexp with a string.

**Comment by chamnap** [comment]
@pocke, Thanks for your quick response 👍 .
